### ISSUE TITLE
scan-build: False positive from core.CallAndMessage

### ISSUE THREAD

**Issue by Tamaranch** [llvm/llvm-project#53617]
Clang 13.0.0.
Code to reproduce the problem:
```c
#include <glib.h>

typedef struct
{
  gchar *str;
} test_struct;


gint main (gint argc, gchar **argv)
{
  test_struct *s1, *s2;
  gchar *str;

  s1 = g_new0 (test_struct, 1);
  s2 = g_memdup (s1, sizeof (test_struct));

  str = g_strconcat ("text", s2->str, NULL);

  g_free (s1);
  g_free (s2);
  g_free (str);
}
```
It triggers this warning:
```
warning: 2nd function call argument is an uninitialized value [core.CallAndMessage]
  str = g_strconcat ("text", s2->str, NULL);
```
This seems to be related to the function name `g_memdup`, because the warning disappears when using the same function declaration with a different name.

**Comment by steakhal** [comment]
Indeed. The `g_memdup` is modeled as if it returns uninitialized memory.

```c++
void MallocChecker::checkGMemdup(const CallEvent &Call,
                                 CheckerContext &C) const {
  ProgramStateRef State = C.getState();
  //                                                vvvvvvvvvvvvvv
  State = MallocMemAux(C, Call, Call.getArgExpr(1), UndefinedVal(), State,
                       AF_Malloc);
  State = ProcessZeroAllocCheck(Call, 1, State);
  C.addTransition(State);
}
```

We could hotfix it by binding `UnknownVal()` instead or actually copy the bindings of the `Call.getArgExpr(1)`.

**Comment by steakhal** [comment]
The fix is under review: https://reviews.llvm.org/D124436
